orbit/pkg/update/notifications.go

```diff
@@ -397,3 +398,94 @@ func (h *runScriptsConfigFetcher) GetConfig() (*fleet.OrbitConfig, error) {
 	}
 	return cfg, err
 }
+
+type DiskEncryptionKeySetter interface {
+	SetOrUpdateDiskEncryptionKey(diskEncryptionStatus fleet.OrbitHostDiskEncryptionKeyPayload) error
+}
+
+type windowsMDMBitlockerConfigFetcher struct {
+	// Fetcher is the OrbitConfigFetcher that will be wrapped. It is responsible
+	// for actually returning the orbit configuration or an error.
+	Fetcher OrbitConfigFetcher
+	// Frequency is the minimum amount of time that must pass between two
+	// executions of the windows MDM enrollment attempt.
+	Frequency time.Duration
+	// HostUUID is the current host's UUID.
+	HostUUID string
+
+	// Bitlocker Operation Results
+	EncryptionResult DiskEncryptionKeySetter
+
+	// tracks last time the enrollment command was executed
+	lastEnrollRun time.Time
+
+	// ensures only one script execution runs at a time
+	mu sync.Mutex
+}
+
+func ApplyWindowsMDMBitlockerFetcherMiddleware(
+	fetcher OrbitConfigFetcher,
+	frequency time.Duration,
+	encryptionResult DiskEncryptionKeySetter,
+) OrbitConfigFetcher {
+	return &windowsMDMBitlockerConfigFetcher{
+		Fetcher:          fetcher,
+		Frequency:        frequency,
+		EncryptionResult: encryptionResult,
+	}
+}
+
+// GetConfig calls the wrapped Fetcher's GetConfig method, and if the fleet
+// server set the "needs windows enrollment" flag to true, executes the command
+// to enroll into Windows MDM (or not, if the device is a Windows Server).
+func (w *windowsMDMBitlockerConfigFetcher) GetConfig() (*fleet.OrbitConfig, error) {
+	cfg, err := w.Fetcher.GetConfig()
+	if err == nil && cfg.Notifications.EnforceBitLockerEncryption {
+		if w.mu.TryLock() {
+			defer w.mu.Unlock()
+
+			w.attemptBitlockerEncryption(cfg.Notifications)
+		}
+	}
+
+	return cfg, err
+}
+
+func (w *windowsMDMBitlockerConfigFetcher) attemptBitlockerEncryption(notifs fleet.OrbitConfigNotifications) {
+	// do not trigger Bitlocker encryption if running on a Windwos server
+	isWindowsServer, err := IsRunningOnWindowsServer()
+	if err != nil || isWindowsServer {
+		log.Error().Msg("skipped calling RegisterDeviceWithManagement to enroll Windows device, device is a server")
+		return
```

Comment:
sorry for not being clear before, I meant: should we do this?

```suggestion
	if err != nil {
		log.Error().Err(err).Msg("checking if the host is a Windows server")
		return
	}
	
	if isWindowsServer {
		log.Debug().Msg("skipped calling RegisterDeviceWithManagement to enroll Windows device, device is a server")
		return
```

Reply:
Thanks, I've just fixed this